test/spawn.jl

```diff
@@ -572,8 +576,15 @@ mktempdir() do dir
             @test Sys.which(foo_path) === nothing
         end
 
+    end
+
+    # Ensure these tests are done only with a PATH of known contents
+    withenv("PATH" => "$(dir)") do
         # Test that completely missing files also return nothing
         @test Sys.which("this_is_not_a_command") === nothing
+
+        # Check that which behaves correctly when passed a blank string
+        @test isnothing(Base.Sys.which(" "))
```

Comment:
I'm a bit late to the party, but should we also test that when there actually is an executable named ` ` in the path we can find it?

Reply:
There are existing tests for that.  I just added a test for a blank string and made sure that the tests that should return nothing are independent of what's on the filesystem.